pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -898,7 +900,36 @@ func (p *Pod) ensureWorkloadOwnedByAllMembers(ctx context.Context, c client.Clie
 	return nil
 }
 
-func (p *Pod) ConstructComposableWorkload(ctx context.Context, c client.Client, r record.EventRecorder) (*kueue.Workload, error) {
+func (p *Pod) getWorkloadLabels(labelKeysToCopy []string) (map[string]string, error) {
+	jobLabels := p.Object().GetLabels()
+	if len(labelKeysToCopy) == 0 {
+		return nil, nil
+	}
+	workloadLabels := make(map[string]string, len(labelKeysToCopy))
+	for _, labelKey := range labelKeysToCopy {
+		if labelValue, found := jobLabels[labelKey]; found {
+			workloadLabels[labelKey] = labelValue
+		}
+	}
```

Comment:
Could we move this pass under the `!p.isGroup` check? It seems the case when this is a group is handled below.

Reply:
This is handling the case when the object is a single pod. We have to copy the labels from the single pod into the workload in this case.

Reply:
Yes, my idea was 
```golang
        workloadLabels := make(map[string]string, len(labelKeysToCopy))
	if !p.isGroup {
	        for _, labelKey := range labelKeysToCopy {
		        if labelValue, found := jobLabels[labelKey]; found {
			        workloadLabels[labelKey] = labelValue
		        }
	        }
		return workloadLabels, nil
	}
        // the code for group handling
```

Reply:
yes, I think its correct.